pandas/core/reshape/merge.py

```diff
@@ -1452,6 +1454,147 @@ def _validate(self, validate: str) -> None:
+        right_index: bool = False,
+        sort: bool = True,
+        suffixes: Suffixes = ("_x", "_y"),
+        copy: bool = True,
+        indicator: bool = False,
+        validate: str | None = None,
+        left_chunk_size: int | None = None,
+        right_chunk_size: int | None = None,
+    ):
+        """
+        Note: Callable `on` function should respect future desired interface.
+        Currently the function "could" make use of df specific logic
+        to return a boolean mask (e.g. left.time.dt.date == right.time.dt.date),
+        but eventually when this implementation is replaced we might not have
+        that ability. Should we (1) restrict those cases now? (2) allow them and
+        fail in the future? or (3) plan to develop future implementation to work
+        with these constructs as well?
+        """
+        self.is_conditional_merge = False
+        if callable(on):
+            if any([left_on, right_on, left_index, right_index]):
+                raise ValueError(
+                    "Cannot define any of (`left_on`, `right_on`, `left_index`, "
+                    "`right_index`) in a conditional merge"
+                )
+
+            if axis != 1:
+                raise ValueError("Conditional merge must use `axis=1`")
+
+            if not copy:
+                # Reason is that due to the initial chunked implementation,
+                # non-trivial to guarantee that copy=False is respected
+                raise ValueError("Conditional merge must use `copy=True`")
+
+            if validate:
+                # Reason is that validation cannot be performed on a chunked basis
+                # TODO: validate on the final result of the chunked merge result?
+                raise NotImplementedError(
+                    "Conditional merge does not support validation"
+                )
+
+            if sort:
+                # Reason is that there is no definitive sort order when using a
+                # custom merge condition
+                raise ValueError("Cannot sort on join keys in a conditional merge")
+
+            if how != "inner":
+                raise NotImplementedError(
+                    '`Conditional merge is currently only available for how="inner". '
+                    "Other merge types will be available in a future version."
+                )
+
+            self.condition = on
+            self.suffixes = suffixes
+            self.indicator = indicator
+
+            # TODO: dynamically determine optimal default chunk size
+            if (not left_chunk_size) or left_chunk_size < 1:
+                left_chunk_size = _DEFAULT_LEFT_CHUNK_SIZE
+
+            if (not right_chunk_size) or right_chunk_size < 1:
+                right_chunk_size = _DEFAULT_RIGHT_CHUNK_SIZE
+
+            left_chunks = _chunks(left, n=left_chunk_size)
+            right_chunks = _chunks(right, n=right_chunk_size)
+            self.chunk_pairs = itertools.product(left_chunks, right_chunks)
```

Comment:
All new attributes should be "private" i.e. `self._`

Reply:
done